File: src/main/java/com/ecommerce/withjava/OrderService.java
package com.ecommerce.withjava;

public class OrderService {

    public double calculateTotal(Order order) {
        double discount = 0.0;

        if (order.getCustomerType().equals("GOLD")) {
            discount = order.getTotalAmount() * 0.20;
        } else if (order.getCustomerType().equals("SILVER")) {
            discount = order.getTotalAmount() * 0.10;
        } else if (order.getCustomerType().equals("BRONZE")) {
            discount = order.getTotalAmount() * 0.05;
        }

        if (order.getTotalAmount() > 1000.0) {
            discount += 50.0;
        }

        double finalValue = order.getTotalAmount() - discount;

        System.out.println("Processando pedido " + order.getId() + " para cliente " + order.getCustomerType());

        return finalValue;
    }
}

File: src/main/kotlin/com/ecommerce/withkotlin/OrderService.kt
package com.ecommerce.withkotlin

class OrderService {

    fun calculateTotal(order: Order): Double {
        var discount = 0.0

        if (order.customerType == "GOLD") {
            discount = order.totalAmount * 0.20
        } else if (order.customerType == "SILVER") {
            discount = order.totalAmount * 0.10
        } else if (order.customerType == "BRONZE") {
            discount = order.totalAmount * 0.05
        }

        if (order.totalAmount > 1000.0) {
            discount += 50.0
        }

        val finalValue = order.totalAmount - discount

        println("Processando pedido ${order.id} para cliente ${order.customerType}")

        return finalValue
    }
}